python/plugins/processing/algs/qgis/Boundary.py

```diff
@@ -78,26 +67,15 @@ def processAlgorithm(self, parameters, context, feedback):
         if QgsWkbTypes.hasM(input_wkb):
             output_wkb = QgsWkbTypes.addM(output_wkb)
 
-        (sink, dest_id) = self.parameterAsSink(parameters, self.OUTPUT_LAYER, context,
-                                               source.fields(), output_wkb, source.sourceCrs())
-
-        features = source.getFeatures()
-        total = 100.0 / source.featureCount() if source.featureCount() else 0
-
-        for current, input_feature in enumerate(features):
-            if feedback.isCanceled():
-                break
-            output_feature = input_feature
-            input_geometry = input_feature.geometry()
-            if input_geometry:
-                output_geometry = QgsGeometry(input_geometry.geometry().boundary())
-                if not output_geometry:
-                    feedback.reportError(self.tr('No boundary for feature {} (possibly a closed linestring?)').format(input_feature.id()))
-                    output_feature.clearGeometry()
-                else:
-                    output_feature.setGeometry(output_geometry)
-
-            sink.addFeature(output_feature, QgsFeatureSink.FastInsert)
-            feedback.setProgress(int(current * total))
-
-        return {self.OUTPUT_LAYER: dest_id}
+        return output_wkb
+
+    def processFeature(self, feature, feedback):
+        input_geometry = feature.geometry()
+        if input_geometry:
+            output_geometry = QgsGeometry(input_geometry.geometry().boundary())
```

Comment:
I think one easy way we can simplify this is by renaming the `QgsAbstractGeometry *QgsGeometry::geometry()` to `QgsAbstractGeometry* QgsGeometry::primitive()`.

So instead of
 
    feature.geometry().geometry().boundary()

it'd be
 
    feature.geometry().primitive().boundary()

It's a bit less ugly...